### PR TITLE
Update CI components in Zcash stack

### PR DESCRIPTION
**Pull request by mdr0id**
Remove legacy CI components, update current CI script and docs. The PR removes much of the old buildbot components to reduce redundant code. Also add details so users can build these images locally.

### CODE REVIEW

File: contrib/ci-builders/tekton/docker-build.sh [hunk 1]
```diff
@@ -10,51 +35,85 @@ docker push electriccoinco/zcashd-build-debian9
...
 docker push electriccoinco/zcashd-worker-ubuntu2004
 
+docker build . -f Dockerfile-tekton-bench --build-arg BASEOS=ubuntu --build-arg FROMBASEOS=ubuntu --build-arg FROMBASEOS_BUILD_TAG=2004 -t electriccoinco/zcashd-bench-ubuntu2004
+docker push electriccoinco/zcashd-bench-ubuntu2004
+
+# Ubuntu 22.04
+docker build . -f Dockerfile-build.apt --build-arg FROMBASEOS=ubuntu --build-arg FROMBASEOS_BUILD_TAG=22.04 -t electriccoinco/zcashd-build-ubuntu2204
+docker push electriccoinco/zcashd-build-ubuntu2204
+
+docker build . -f Dockerfile-tekton-worker --build-arg BASEOS=ubuntu --build-arg FROMBASEOS=ubuntu --build-arg FROMBASEOS_BUILD_TAG=2204 -t electriccoinco/zcashd-worker-ubuntu2004
```
Comment (str4d) [inline contrib/ci-builders/tekton/docker-build.sh:95]: Why do the build and worker/bench lines have different `FROMBASEOS_BUILD_TAG` values (both for these new ones, and for all the existing ones)? The build variant has an additional `.` present.

**Review by str4d — DISMISSED** [review]
utACK 3a2fe473bd4845573efee76d7e5fb577b3899b02 with non-blocking comments and a question.

File: contrib/ci-builders/tekton/Dockerfile-tekton-bench [hunk 2]
```diff
@@ -0,0 +1,9 @@
+ARG FROMBASEOS
+ARG FROMBASEOS_BUILD_TAG
+FROM electriccoinco/zcashd-worker-$FROMBASEOS$FROMBASEOS_BUILD_TAG
+
+COPY ./benchmark-200k-UTXOs.tar.xz /home/benchmark-200k-UTXOs.tar.xz
+COPY ./block-107134.tar.xz /home/block-107134.tar.xz
```
[the remark this answers is not in the document]
Comment (mdr0id) [reply contrib/ci-builders/tekton/Dockerfile-tekton-bench:6]: Oh, are there specific heights or generation params recommended ? Most of this I wrote just from the legacy benchmark suite.

File: contrib/ci-builders/tekton/docker-build.sh (hunk 1 above)
Comment (mdr0id) [reply contrib/ci-builders/tekton/docker-build.sh:95]: I added in the missing docker files that some how got nuked on merging some branches. They are used to layer the builds per update but newer conventions might not need these injected as ARG.

The . at the end of the docker build command tells Docker that it should look for the Dockerfile in the current directory. IIRC, this is also an older convention.

File: contrib/ci-builders/tekton/Dockerfile-tekton-bench (hunk 2 above)
[the remark this answers is not in the document]
Comment (str4d) [reply contrib/ci-builders/tekton/Dockerfile-tekton-bench:8]: The script pins the relevant block heights, and will generate the files. I can also just send them to you directly if you prefer.

File: contrib/ci-builders/tekton/docker-build.sh (hunk 1 above)
Comment (str4d) [reply contrib/ci-builders/tekton/docker-build.sh:95]: What I meant is that one of the lines uses `22.04` and the others use `2204`, for the same configuration option (which from its name sounds like a git tag, so I'd expect only one of the two variants to be valid).

File: contrib/ci-builders/tekton/Dockerfile-build.centos8
Comment (nuttycom) [inline contrib/ci-builders/tekton/Dockerfile-build.centos8:1]: We're no longer supporting centos8, so this can probably be deleted.

**Comment by mdr0id** [comment]
Closing since this branch is now a mess. Please use #6485

**Pull request closed without merging**